src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
```

Comment:
First off, when  you move your shortened return statement in arrow function to a newline, use round brackets:

```
    : people.filter(p => people.some(m => (
         m.name === p.mother
      ));
```

But there's even another issue. This ternary operator might be hard to understand. The code could be simplified by using a single filter operation with a more concise condition.
Take a look at this:
```
  const children = people.filter((person) => {
    const isChild = people.some(
      (innerPerson) => innerPerson.mother === person.name
    )
    const isSon = onlyWithSon && person.sex === 'm'

    return isChild && (!onlyWithSon || isSon)
  })
```
In this version, the filtering logic is combined into a single filter operation. The condition in the filter callback function checks whether the person is a child (i.e., has a mother in the people array) and whether the `onlyWithSon` parameter is true and the person is a son. If `onlyWithSon` is false, the second part of the condition is always true, so all children are kept. If `onlyWithSon` is false, the second part of the condition is always true, so all children are kept. If `onlyWithSon` is true, only male children with a mother in the people array are kept. This simplifies the code and eliminates the need for two separate filter operations.